airflow/utils/db.py

```diff
@@ -1240,19 +1247,20 @@ class BadReferenceConfig:
         bad_rows_subquery = bad_ref_cfg.exists_func(session, source_table, **exists_func_kwargs)
         select_list = [x.label(x.name) for x in source_table.c]
         invalid_rows_query = session.query(*select_list).filter(~bad_rows_subquery.exists())
-        invalid_row_count = invalid_rows_query.count()
-        if invalid_row_count <= 0:
-            continue
 
         dangling_table_name = _format_airflow_moved_table_name(source_table.name, change_version, 'dangling')
         if dangling_table_name in existing_table_names:
-            yield _format_dangling_error(
-                source_table=source_table.name,
-                target_table=dangling_table_name,
-                invalid_count=invalid_row_count,
-                reason=f"without a corresponding {bad_ref_cfg.ref_table} row",
-            )
-            errored = True
+            invalid_row_count = invalid_rows_query.count()
```

Comment:
Can we do a simple count against the `moved` table instead?

Reply:
that would tell us what is in the moved table that's already there -- not what remains in the table -- if the rows are already purged, we can proceed, even though the "moved" table is there

Reply:
Ah, yeah, I see now.

Reply:
The moved table can't be there already -- either the migration has already applied (and we created the moved table) or it's not yet applied and the table can't yet exist.

But once the table exists we can't have anymore bad rows as there is a FK enforcing it.

Am I missing something?

Reply:
It's not gonna be there in normal circumstances.  but it could be there if user downgrades and upgrades again, or if the upgrade failed after pre-migration checks but before migration completed and user is trying to upgrade again.  FWIW this check pre-dated my involvement.  i think it's just a defensive "whoa something weird going on here you better deal with this table first".

In this [draft pr](https://github.com/apache/airflow/pull/23485) (not ready for review) i explored just appending rows in this circumstance.